spark3/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java

```diff
@@ -156,12 +170,20 @@ public PartitionReaderFactory createReaderFactory() {
 
     boolean hasNoDeleteFiles = tasks().stream().noneMatch(TableScanUtil::hasDeletes);
 
+    boolean hasTimestampWithoutZone = hasTimestampWithoutZone(expectedSchema);
+
     boolean readUsingBatch = batchReadsEnabled && hasNoDeleteFiles && (allOrcFileScanTasks ||
-        (allParquetFileScanTasks && atLeastOneColumn && onlyPrimitives));
+            (allParquetFileScanTasks && atLeastOneColumn && onlyPrimitives && !hasTimestampWithoutZone));
 
     return new ReaderFactory(readUsingBatch ? batchSize : 0);
   }
 
+  private static boolean hasTimestampWithoutZone(Schema schema) {
```

Comment:
Another candidate for the public utility class

Reply:
addressed it